File: Day3/main.ts
type Inventory = Array<{name: string;quantity: number;category: string;}>;

function organizeInventory(inventory: Inventory): Record<string, Record<string, number>> {
  const result: Record<string, Record<string, number>> = {};
  for (const { name, quantity, category } of inventory) {
    result[category] ??= {};
    result[category][name] = (result[category][name] || 0) + quantity;
  }
  return result;
}

File: Day3/main.py
def organize_inventory(inventory):
    organized = {}
    for item in inventory:
        category, name, quantity = item['category'], item['name'], item['quantity']
        organized[category] = organized.get(category, {})
        organized[category][name] = organized[category].get(name, 0) + quantity
    return organized
